core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java

```diff
@@ -338,9 +338,17 @@ private void onTopologyEvent(TopologyEvent event) {
       if (event.type == TopologyEvent.Type.SUGGEST_UP) {
         Node node = context.getMetadataManager().getMetadata().getNodes().get(event.address);
         if (node.getDistance() != NodeDistance.IGNORED) {
-          LOG.debug(
-              "[{}] Received a SUGGEST_UP event for {}, reconnecting pool now", logPrefix, node);
-          createOrReconnectPool(node);
+          if (pending.containsKey(node)) {
+            LOG.debug(
+                "[{}] Received a SUGGEST_UP event for {} while pool was already initializing, "
+                    + "ignoring",
+                logPrefix,
+                node);
+          } else {
+            LOG.debug(
+                "[{}] Received a SUGGEST_UP event for {}, reconnecting pool now", logPrefix, node);
+            createOrReconnectPool(node);
+          }
```

Comment:
So the goal here is to handle this case (from `TopologyEvent.suggestUp()` javadocs):
> if the driver was currently reconnecting to the node, this causes the current ReconnectionSchedule to be reset, and the next reconnection attempt to happen immediately.

The bug is that it calls a method that _also creates the pool_ if it doesn't exist. We don't want to create, only reconnect if it exists. So I would just replace the call to `createOrReconnectPool` with the relevant portion of its body:
```java
ChannelPool pool = pools.get(node);
if (pool != null) {
  pool.reconnectNow();
}
```
Once you do that, you don't really need to check `pending` anymore, because when a pool is still pending (i.e. initializing), it's not in `pools` yet.

PS: there's a small race condition because the pool could start reconnecting immediately after it's initialized, before we put it in `pools` (see `onPoolReady`). If a `SUGGEST_UP` event also arrives exactly at this moment, we could miss it and not trigger a reconnection. But it doesn't really matter because regular scheduled reconnections will eventually kick in, and it's the beginning of the schedule so the interval should be pretty short.

Reply:
yeah, that seems like a better solution and more resilient to problems in the future if other methods call `createOrReconnectPool` without first checking pending.

Reply:
Ah I misread, you are suggesting to just replace the call to `createOrReplacePool` with simply calling `pool.reconnectNow()`, got it.